Packs/EmailCommunication/Scripts/SendEmailReply/SendEmailReply.py

```diff
@@ -222,12 +459,436 @@ def get_mailbox_from_incident_labels(labels):
+    for idx, thread_entry in enumerate(incident_email_threads):
+        if str(thread_entry['EmailCommsThreadNumber']) == str(email_selected_thread):
+            thread_found = True
+
+            if thread_entry['MessageDirection'] == 'inbound':
+                reply_to_message_id = thread_entry['MessageID']
+                outbound_only = False
+            reply_code = thread_entry['EmailCommsThreadId']
+            reply_subject = thread_entry['EmailSubject']
+
+            # Create recipient list based on all 'EmailTo' and 'EmailFrom' values
+            if thread_entry['EmailTo'] not in reply_recipients:
+                if len(reply_recipients) == 0:
+                    reply_recipients = thread_entry['EmailTo']
+                else:
+                    reply_recipients += f", {thread_entry['EmailTo']}"
+
+            if thread_entry['EmailFrom'] not in reply_recipients:
+                if len(reply_recipients) == 0:
+                    reply_recipients = thread_entry['EmailFrom']
+                else:
+                    reply_recipients += f", {thread_entry['EmailFrom']}"
+
+            # Create list of mailboxes receiving messages in this thread in case there are multiple
+            if thread_entry['EmailReceived'] not in reply_mailbox:
+                if len(reply_mailbox) == 0:
+                    reply_mailbox = thread_entry['EmailReceived']
+                else:
+                    reply_mailbox += f", {thread_entry['EmailReceived']}"
+
+            # Create list of CC addresses based on others CC'd on this thread
+            if thread_entry['EmailCC']:
+                for cc_address in thread_entry['EmailCC'].split(","):
+                    if cc_address not in thread_cc and len(thread_cc) == 0:
+                        thread_cc = cc_address
+                    elif cc_address not in thread_cc:
+                        thread_cc += f',{cc_address}'
+
+            # Create list of BCC addresses based on others BCC'd on this thread
+            if thread_entry['EmailBCC']:
+                for bcc_address in thread_entry['EmailBCC'].split(","):
+                    if bcc_address not in thread_bcc and len(thread_bcc) == 0:
+                        thread_bcc = bcc_address
+                    elif bcc_address not in thread_bcc:
+                        thread_bcc += f',{bcc_address}'
+            # Keep track of the last processed list position
+            last_thread_processed = idx
+
+    return thread_found, reply_to_message_id, outbound_only, reply_code, reply_subject, reply_recipients,\
+        reply_mailbox, thread_cc, thread_bcc, last_thread_processed
+
+
+def multi_thread_reply(new_email_body, incident_id, email_selected_thread, new_email_attachments, files, add_cc,
+                       add_bcc, service_mail, mail_sender_instance, new_attachment_names):
+    """
+    Retrieve all entries in the EmailThreads context key
```

Comment:
this is the same description as `collect_thread_details`. please add accurate and informative description.

Reply:
Done.